continuous_integration/scripts/test_report.py

```diff
@@ -140,14 +140,10 @@ def get_jobs(run, session):
     # Somehow the job ID is not part of the workflow schema and we have no other way to later join
     # this to the JXML results.
 
-    name_components = (
-        df_jobs.name.str.extract(r"test \((.+)\)", expand=False)
-        .dropna()
-        .str.split(", ", expand=True)
-    )
-    if len(name_components.columns) != 4:
-        raise ValueError(f"Job names must have 4 components:\n{name_components!r}")
-    name_components.columns = ["OS", "environment", "label", "partition"]
+    name_components = df_jobs.name.str.split("-", expand=True).dropna()
+    if len(name_components.columns) != 5:
+        raise ValueError(f"Job names must have 5 components:\n{name_components!r}")
+    name_components.columns = ["OS", "OS version", "environment", "label", "partition"]
```

Comment:
The new naming schema in https://github.com/dask/distributed/blob/77591e9be578159fa41ea969ce5fd5a24ad3d8c9/.github/workflows/tests.yaml#L20 means that we now separate the OS name and its version (.e.g, `windows-latest` will be split into `windows` and `latest`).

Reply:
I suggest to add a comment in `tests.yaml` that indicates we should adjust the test report case. Having this check is fine but if we can avoid this in the first place, that's better. Something like

```yaml
# If you change the pattern of this name, please adjust the scripts/test_report.py parsing accordingly
name: ${{ matrix.os }}-${{ matrix.environment }}-${{ matrix.label }}-${{ matrix.partition }}
```

Reply:
Done.